packages/core/src/transport/batch.ts

```diff
@@ -11,6 +11,14 @@ export interface BatchFlushEvent {
   bufferMessagesCount: number
 }
 
+export type FlushReason =
+  | 'batch_duration_limit'
+  | 'batch_bytes_limit'
+  | 'before_unload'
+  | 'pagehide'
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏`page_hide`, to match other reasons pattern